src/parser/lval.js

```diff
@@ -378,4 +380,18 @@ export default class LValParser extends NodeUtils {
       }
     }
   }
+
+  checkToRestConversion(node: SpreadElement): void {
+    const validArgumentTypes = [
+      "Identifier",
+      "MemberExpression",
+      "AssignmentPattern",
+    ];
+
+    if (validArgumentTypes.indexOf(node.argument.type) !== -1) {
```

Comment:
imho `validArgumentTypes` is more readable as its describing in the name why we check those and this dont have to be deduced by the reader, but I can refactor it to `&&` if u want